chain/app/app.go

```diff
@@ -395,6 +374,32 @@ func (app *bandApp) DeliverTx(req abci.RequestDeliverTx) (res abci.ResponseDeliv
 	return response
 }
 
+// EndBlocker application updates every end block
+func (app *bandApp) EndBlocker(ctx sdk.Context, req abci.RequestEndBlock) abci.ResponseEndBlock {
+	return app.mm.EndBlock(ctx, req)
+}
+
+func (app *bandApp) Commit() (res abci.ResponseCommit) {
+	app.DeliverContext = sdk.Context{}
+	return app.BaseApp.Commit()
+}
+
+// InitChainer application update at chain initialization
+func (app *bandApp) InitChainer(ctx sdk.Context, req abci.RequestInitChain) abci.ResponseInitChain {
+	var genesisState simapp.GenesisState
+	app.cdc.MustUnmarshalJSON(req.AppStateBytes, &genesisState)
+
+	res := app.mm.InitGenesis(ctx, app.cdc, genesisState)
+
+	// Set Historical infos in InitChain to ignore genesis params
+	stakingParams := staking.DefaultParams()
+	stakingParams.HistoricalEntries = 1000
```

Comment:
db app things? Have related logic in our chain?

Reply:
This is not db app things. It's from here https://github.com/cosmos/gaia/blob/ibc-alpha/app/app.go#L328. 

Now that I look at it again. It's `gaia`'s IBC hack to store old validator set. I think we should leave it here for now. We can move it to default genesis param later. We can talk more if you are still not convinced.